src/Adapter/InMemoryAdapter.php

```diff
@@ -163,18 +168,22 @@ public function loadEvents(
 
     public function loadEventsReverse(
         StreamName $streamName,
-        array $metadata = [],
         int $fromNumber = 0,
-        int $count = null
+        int $count = null,
+        MetadataMatcher $metadataMatcher = null
     ): Iterator {
         if (! isset($this->streams[$streamName->toString()])) {
```

Comment:
I would prefer a hasStream method for readability. Maybe with a StreamCollection.